### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by xepozz**

### CODE REVIEW

File: src/CodeFileSaver.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Gii;
+
+use Exception;
+use ReflectionException;
+use Yiisoft\Yii\Gii\Exception\InvalidConfigException;
+use Yiisoft\Yii\Gii\Generator\AbstractGeneratorCommand;
+
+class CodeFileSaver
```
Comment (samdark) [inline src/CodeFileSaver.php:12]:
```suggestion
class CodeFileWriter
```

File: src/CodeFileSaver.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+     * @param CodeFile[] $files the code files to be saved
+     * @param string[] $results this parameter receives a value from this method indicating the log messages
+     * generated while saving the code files.
+     *
+     * @return bool whether files are successfully saved without any error.
+     * @throws InvalidConfigException
+     *
+     * @throws ReflectionException
+     */
+    public function save(AbstractGeneratorCommand $command, array $files, array $answers, array &$results): bool
```
Comment (samdark) [inline src/CodeFileSaver.php:26]:
```suggestion
    public function write(AbstractGeneratorCommand $command, array $files, array $answers, array &$results): bool
```

File: src/CodeFileSaver.php (hunk 1 above)
Comment (samdark) [inline src/CodeFileSaver.php:12]:
```suggestion
class CodeFileWriter
```

Comment (samdark) [reply src/CodeFileSaver.php:12]: final?

File: src/Controller/DefaultController.php [hunk 3]
```diff
@@ -20,55 +23,47 @@ public function __construct(private DataResponseFactoryInterface $responseFactor
...
-            'name' => $generator->getName(),
-            'description' => $generator->getDescription(),
-            'template' => $generator->getTemplate(),
-            'templatePath' => $generator->getTemplatePath(),
-            'templates' => $generator->getTemplates(),
+            'name' => $generator::getName(),
+            'description' => $generator::getDescription(),
+            'commandClass' => $generator::getCommandClass(),
+            //            'templatePath' => $generator->getTemplatePath(),
+            //            'templates' => $generator->getTemplates(),
```
Comment (samdark) [inline src/Controller/DefaultController.php:40]: ?

**Comment by xepozz** [comment]
Generating a controller and a view works now both via console and http

File: src/Controller/DefaultController.php (hunk 3 above)
Comment (xepozz) [reply src/Controller/DefaultController.php:40]: I'll fix it later or remove if it is not needed

File: src/CodeFileWriter.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+     * @param CodeFile[] $files the code files to be saved
+     * @param string[] $results this parameter receives a value from this method indicating the log messages
+     * generated while saving the code files.
+     *
+     * @throws InvalidConfigException
+     * @throws ReflectionException
+     *
+     * @return bool whether files are successfully saved without any error.
+     */
+    public function write(AbstractGeneratorCommand $command, array $files, array $answers, array &$results): bool
```
Comment (rustamwin) [inline src/CodeFileWriter.php:26]:
```suggestion
    public function write(GeneratorCommandInterface $command, array $files, array $answers, array &$results): bool
```

**Review by rustamwin — CHANGES_REQUESTED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**